### PR TITLE
FIX Check GRUB install_device's for inaccessible files (device)

### PR DESCRIPTION
**Pull request by umtk-webpros**
PAUX-6367: Distupgrade tool does not handle issue with GRUB misconfiguration

### CODE REVIEW

File: pleskdistup/actions/grub.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+from pleskdistup.common import action, dist, util
+
+DEBCONF_CMD = "/usr/bin/debconf-show"
+
+class AssertGrubInstallDeviceExists(action.CheckAction):
+    def __init__(self) -> None:
+        self.name = "check GRUB installation device exists"
+        self.description = """Grub's install-device is not found: {}
+This could fail conversion - please fix it.
```
Comment (SandakovMM) [inline pleskdistup/actions/grub.py:12]: Can we propose a solution to fix this problem here, ideally using a bunch of CLI commands?

File: pleskdistup/actions/grub.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+"""
+
+    def _do_check(self) -> bool:
+        if not dist.get_distro().deb_based:
+            return True
+
+        debconf_out = util.logged_check_call([DEBCONF_CMD, "grub-pc"])
+        prefix = "grub-pc/install_devices:"
```
Comment (SandakovMM) [inline pleskdistup/actions/grub.py:20]: NIT: We could turn this into a class member and make it configurable through the init function argument. However, it does not seem critical at the moment.

File: pleskdistup/actions/grub.py [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+        for line in debconf_out.split("\n"):
+            if line.find(prefix) == 0:
+                devpath = line[len(prefix):].strip()
+                if not os.path.exists(devpath):
```
Comment (SandakovMM) [inline pleskdistup/actions/grub.py:24]: Is it possible to have multiple `grub-pc/install_devices:` lines in the output?

Comment (ukablan-wpc) [reply pleskdistup/actions/grub.py:27]: The virtual instance I encountered had only one device path. Although the plural noun suspects of multi device but I could not find a way to make it.

File: pleskdistup/actions/grub.py [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+    def __init__(self) -> None:
+        self.name = "check GRUB installation device exists"
+        self.description = """Grub's install-device is not found: {}
+This could fail conversion - please fix GRUB configuration:
+    dpkg --configure grub-pc
+"""
+
+    def _do_check(self) -> bool:
+        if not os.path.exists(DEBCONF_CMD):
+            return True
```
Comment (ukablan-wpc) [inline pleskdistup/actions/grub.py:18]: Better to check distro IMO

File: pleskdistup/actions/grub.py (hunk 1 above)
Comment (ukablan-wpc) [reply pleskdistup/actions/grub.py]: I added 
```
dpkg --configure grub-pc
```

File: pleskdistup/actions/grub.py (hunk 2 above)
Comment (ukablan-wpc) [reply pleskdistup/actions/grub.py]: Yes, it seems like (to stay) very debian-specific class anyway.

**Review by SandakovMM — APPROVED** [review]

**Pull request merged**